server/courselist.rb

```diff
@@ -124,17 +124,25 @@ class CourseList < Sinatra::Base
   ## This method will return the contents of the requested (or default) configuration file.
   ## Methods in a Sinatra module need to be defined in a helpers section.
   helpers do
-    def self.get_local_config_yml(requested_file, default_file)
-      if File.exist? requested_file
-        file_name = requested_file
+
+    def self.verify_file_is_usable(requested_file)
+      return requested_file if ((File.exists? requested_file) && File.readable?(requested_file))
+      nil
+    end
+
+    def self.get_local_config_yml(requested_file, default_file, required)
+
+      file_name = verify_file_is_usable(requested_file) || verify_file_is_usable(default_file) || nil
+
+      if file_name.nil? then
+        logger.fatal "can not find requested or default configuration file: [#{requested_file}] or [#{default_file}]" if required
```

Comment:
if get_local_config_yml return value, what will be return here if require is false?